### PR TITLE
Improve missing default signer error messaging

### LINKED ISSUE #17423: Improve `solana program deploy` error messag when default keypair does not exist
#### Problem

Using the Anchor project Dockerfile as a template to download solana onto my server.

Given I download version v1.6.9,
when I compile my contract and attempt to deploy it to a local testnet by calling `solana program deploy <path>`
then I get os error 2.

This works in v1.6.6

v1.6.9
```
$ solana program deploy /path/to/shared_object.so
Error: Dynamic program error: No such file or directory (os error 2)
$ solana --version
solana-cli 1.6.9 (src:9e42883d; feat:2960423209)
$ anchor --version
anchor-cli 0.6.0
$ solana program deploy
Error: Dynamic program error: No such file or directory (os error 2)
```

v1.6.6
```
$ solana program deploy /path/to/shared_object.so
Program Id: 9A7oBfLrXY8QACcBP4CBzu2Z52eqULMT8cvA5KVnqMW
$ solana --version
solana-cli 1.6.6 (src:fe775a97; feat:3714435735)
$ anchor --version
anchor-cli 0.6.0
$ solana program deploy
Error: Program location required if buffer not supplied
```

#### Proposed Solution
Revert the changes made on this subcommand in between v1.6.6 and v1.6.9

### PR DESCRIPTION
**Pull request by jackcmay**
#### Problem

The CLI returns a vauge error message if the default signer is not present

#### Summary of Changes

Require that a default signer be present and report a better error message:

```
$ solana address
Error: No default signer found, run "solana-keygen new -o /Users/jack/.config/solana/id.json" to create a new one
```

Fixes #17423

### CODE REVIEW

File: clap-utils/src/keypair.rs [hunk 1]
```diff
@@ -316,7 +337,7 @@ pub fn signer_from_path_with_config(
         legacy,
     } = parse_signer_source(path)?;
     match kind {
-        SignerSourceKind::Prompt => {
+SignerSourceKind::Prompt => {
```
Comment (t-nelson) [inline clap-utils/src/keypair.rs:340]: How'd you sneak this past `rustfmt`? :slightly_smiling_face:

File: clap-utils/src/keypair.rs [hunk 2]
```diff
@@ -257,6 +277,7 @@ pub(crate) fn parse_signer_source<S: AsRef<str>>(
                     ASK_KEYWORD => Ok(SignerSource::new_legacy(SignerSourceKind::Prompt)),
                     _ => match Pubkey::from_str(source.as_str()) {
                         Ok(pubkey) => Ok(SignerSource::new(SignerSourceKind::Pubkey(pubkey))),
+                        // Err(_) => Ok(SignerSource::new(SignerSourceKind::Filepath(source))),
```
Comment (t-nelson) [inline clap-utils/src/keypair.rs:280]: Delete?

**Review by t-nelson — DISMISSED** [review]
lgtm with nits

File: clap-utils/src/keypair.rs (hunk 1 above)
Comment (jackcmay) [reply clap-utils/src/keypair.rs:340]: Apparently `rustfmt` has the same blind spot I do :-)

**Pull request merged**